include/nuttx/syslog/syslog.h

```diff
@@ -84,12 +84,17 @@
 
 /* This structure provides the interface to a SYSLOG device */
 
-typedef CODE ssize_t (*syslog_write_t)(FAR const char *buf, size_t buflen);
-typedef CODE int (*syslog_putc_t)(int ch);
-typedef CODE int (*syslog_flush_t)(void);
+typedef CODE ssize_t (*syslog_write_t)(FAR void *dev, FAR const char *buf,
```

Comment:
should we pass ```FAR struct syslog_channel_s *``` instead

Reply:
I was thinking that it is better to keep it as a `void *` temporarily, as the underlying channels are not uniform yet (like ARM ITM for example).

But I think I will just do it properly, and go for your your recommendation above.

Reply:
My suggestion come from other drivers(e.g. spi_dev_s/spi_ops_s) practice. It's better to do the right interface design in the first step, instead changing the interfae prootype again and again.

Reply:
@xiaoxiang781216 You may have a look, it seems better now.

I think this is what you had in mind.